onaio/onadata
# Submissions and Forms Notifications

### Changes / Features implemented

Send MQTT message to a channel upon Submission Review and Submission Deletion.


### Steps taken to verify this change does what is intended
- Written tests

### To test
1. MQTT server
- ssh to mqtt server
- listen to any of the channels below
- perform the relevant action (delete, create, edit ...)
- a message payload should be posted on the channel

2. API
- perform the relevant action (delete, create, edit ...)
- a message payload should be included in the relevant API json response 

|channel | url| action|
|---------|--|--------|
|`/onadata/xform/[form_id]/submission_deleted`| `/api/v1/messaging.json?target_type=xform&target_id={form_id}&verb=submission_deleted`|submission deletion|
|`/onadata/xform/[form_id]/submission_created `|`/api/v1/messaging.json?target_type=xform&target_id={form_id}&verb=submission_created`| submission creation |
|`/onadata/xform/[form_id]/submission_edited `| `/api/v1/messaging.json?target_type=xform&target_id={form_id}&verb=submission_edited`|submission edit |
|`/onadata/xform/[form_id]/submission_reviewed `| `/api/v1/messaging.json?target_type=xform&target_id={form_id}&verb=submission_review`|submission_review|
|`/onadata/xform/[form_id]/form_updated `| `/api/v1/messaging.json?target_type=xform&target_id={form_id}&verb=form_updated`|form updated|

Example payload:
```json
{
"id": 113, 
"verb": 
"submission_deleted", 
"message": {"id": ["35838"]}, 
"user": "linc", 
"timestamp": "2020-06-10T11:10:45.346353+00:00"
}
```




Closes #1791
